subsys/shell/backends/shell_mqtt.c

```diff
@@ -124,9 +124,12 @@ static int wait(int timeout)
 static int get_mqtt_broker_addrinfo(void)
 {
 	int rc;
-	struct zsock_addrinfo hints = { .ai_family = AF_INET,
-					.ai_socktype = SOCK_STREAM,
-					.ai_protocol = 0 };
+	struct zsock_addrinfo hints;
+
+	(void)memset(&hints, 0, sizeof(hints));
```

Comment:
I might be missing something, but is this change actually fixing anything? As `hints` structure has an initializer, remainig, uninitialized fields should be zeroed anyway.

Reply:
The variable is on stack, so if not `memset` explicitly I guess there might be some random values for the remaining fields?

Reply:
That is true only if there were no initialization at all, then a stack variable would contain rubbish. But if there is an initializer (may even be empty), all of the remaining fields are zeroed, it's guaranteed by the standard.

Reply:
Interesting, that's something new for me, I will change them